### PR TITLE
Add error message for empty allow credentials when authenticating

### PR DESCRIPTION
**Pull request by weiilug**

### CODE REVIEW

File: src/public/scripts/main.ts [hunk 1]
```diff
@@ -552,6 +552,10 @@ const onAuthenticate = async (): Promise<void> => {
   loading.start();
   const opts = <WebAuthnAuthenticationObject>collectOptions('authentication');
   try {
+    // throw errors when the allow credential list is empty when discoverable credential is not required
+    if (opts?.credentialsToAllow?.length == 0 && $('#resident-key').value !== 'required' ) {
```
Comment (agektmr) [inline src/public/scripts/main.ts:556]: Why do you need this? There shouldn't be a case where `#resident-key` is respected with empty `allowCredentials`. See https://github.com/google/webauthndemo/blob/main/src/public/scripts/main.ts#L176 . If you see the error, the bug should be somewhere else.

Comment (weiilug) [reply src/public/scripts/main.ts:556]: I think we agree with the following rules:
1. when the resident key is enabled, empty allow credential is permitted. This is okay.
2. when in a normal situation (resident key is not enabled), empty allow credentials should raise an exception.

I think from a user's perspective, we shouldn't provide them the option to create empty allow credentials for non-resident-key situation. But as a demo website, we should provide the test team the corner case, so that they can in turn test it.

Comment (agektmr) [reply src/public/scripts/main.ts:556]: I mean, `resident-key` option is only used on registration and `allowCredentials` options is only used on authentication. They won't appear in the same options.  
Do you mean the test team wants to try "when an authenticator is registered without `resident-key`, sending an empty `allowCredentials` should return an error"?

Comment (weiilug) [reply src/public/scripts/main.ts:556]: Actually, https://github.com/google/webauthndemo/blob/a15481db92a818052f04f259d3cc10ab7935074f/src/public/scripts/main.ts#L411 will deal with this situation. But you will find it prints the same message as the case that the security key is not found. But I believe the former should have a higher exception level.

Comment (weiilug) [reply src/public/scripts/main.ts:556]: Then I understand. It's sure not correct to do this. Will cancel this request.
